pkg/storage/stores/shipper/testutil/testutil.go

```diff
@@ -142,7 +147,28 @@ func SetupDBTablesAtPath(t *testing.T, tableName, path string, dbs map[string]DB
 
 	for name, dbRecords := range dbs {
 		AddRecordsToDB(t, filepath.Join(tablePath, name), boltIndexClient, dbRecords.Start, dbRecords.NumRecords)
+		if compressRandomFiles && rand.Intn(2) == 0 {
```

Comment:
In general, I think adding randomness like this to tests is bad practice because it creates flakey test cases. Instead, we can run the test case twice, once against compressed files and once against uncompressed files, or you can make this "randomness" deterministic by basing it off of the test key or index like
```go
if compress && hash(key) % 2 == 0 {
  // compress it
}
```

Reply:
Fixed it. Thanks for the review!